service/controller/chart/v1/resource/release/resource.go

```diff
@@ -111,22 +111,32 @@ func (r *Resource) updateAnnotations(ctx context.Context, cr v1alpha1.Chart, rel
 	currentChecksum := key.ValuesMD5ChecksumAnnotation(*currentCR)
 
 	if releaseState.ValuesMD5Checksum != currentChecksum {
-		if currentCR.Annotations != nil {
-			annotations = currentCR.Annotations
+		annotations := map[string]string{
+			key.ValuesMD5ChecksumAnnotationName: releaseState.ValuesMD5Checksum,
+		}
+
+		patches := []Patch{
```

Comment:
@giantswarm/team-batman This is wrong, it would replace the whole annotation (checked on ginger CP), not just `md5sum`, to do it correctly we need to use `add` operation.